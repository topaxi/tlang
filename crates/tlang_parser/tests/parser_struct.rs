mod common;

#[test]
fn test_parse_struct_definition() {
    assert_parser_snapshot!("struct Foo { x: Int, y: Int }");
}
