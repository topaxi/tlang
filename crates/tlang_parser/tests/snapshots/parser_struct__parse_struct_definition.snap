---
source: crates/tlang_parser/tests/parser_struct.rs
expression: "parse! (\"struct Foo { x: Int, y: Int }\")"
---
Module(
  id: 2,
  statements: [
    Stmt(
      id: 7,
      kind: StructDeclaration(StructDeclaration(
        name: Ident(
          name: "Foo",
          span: Span(
            start: LineColumn(
              line: 0,
              column: 7,
            ),
            end: LineColumn(
              line: 0,
              column: 10,
            ),
          ),
        ),
        fields: [
          StructField(
            id: 4,
            name: Ident(
              name: "x",
              span: Span(
                start: LineColumn(
                  line: 0,
                  column: 13,
                ),
                end: LineColumn(
                  line: 0,
                  column: 14,
                ),
              ),
            ),
            ty: Ty(
              id: 3,
              name: Path(
                segments: [
                  Ident(
                    name: "Int",
                    span: Span(
                      start: LineColumn(
                        line: 0,
                        column: 16,
                      ),
                      end: LineColumn(
                        line: 0,
                        column: 19,
                      ),
                    ),
                  ),
                ],
                res: Res(
                  node_id: None,
                ),
                span: Span(
                  start: LineColumn(
                    line: 0,
                    column: 16,
                  ),
                  end: LineColumn(
                    line: 0,
                    column: 19,
                  ),
                ),
              ),
              parameters: [],
              span: Span(
                start: LineColumn(
                  line: 0,
                  column: 0,
                ),
                end: LineColumn(
                  line: 0,
                  column: 0,
                ),
              ),
            ),
          ),
          StructField(
            id: 6,
            name: Ident(
              name: "y",
              span: Span(
                start: LineColumn(
                  line: 0,
                  column: 21,
                ),
                end: LineColumn(
                  line: 0,
                  column: 22,
                ),
              ),
            ),
            ty: Ty(
              id: 5,
              name: Path(
                segments: [
                  Ident(
                    name: "Int",
                    span: Span(
                      start: LineColumn(
                        line: 0,
                        column: 24,
                      ),
                      end: LineColumn(
                        line: 0,
                        column: 27,
                      ),
                    ),
                  ),
                ],
                res: Res(
                  node_id: None,
                ),
                span: Span(
                  start: LineColumn(
                    line: 0,
                    column: 24,
                  ),
                  end: LineColumn(
                    line: 0,
                    column: 27,
                  ),
                ),
              ),
              parameters: [],
              span: Span(
                start: LineColumn(
                  line: 0,
                  column: 0,
                ),
                end: LineColumn(
                  line: 0,
                  column: 0,
                ),
              ),
            ),
          ),
        ],
      )),
      span: Span(
        start: LineColumn(
          line: 0,
          column: 0,
        ),
        end: LineColumn(
          line: 0,
          column: 29,
        ),
      ),
      leading_comments: [],
      trailing_comments: [],
    ),
  ],
  span: Span(
    start: LineColumn(
      line: 0,
      column: 0,
    ),
    end: LineColumn(
      line: 0,
      column: 29,
    ),
  ),
)
